In [3]:
import os

os.chdir('../')

In [4]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

data_train_path = "Dataset/data_train.csv"

URM_all_dataframe = pd.read_csv(data_train_path)
print(URM_all_dataframe.head(10))

   user_id  item_id  data
0        0        0   1.0
1        0        2   1.0
2        0      120   1.0
3        0      128   1.0
4        0      211   1.0
5        0      232   1.0
6        0      282   1.0
7        0      453   1.0
8        0      458   1.0
9        0      491   1.0


In [5]:
import numpy as np
import scipy.sparse as sps
from Data_manager.split_functions.split_train_validation_random_holdout import split_train_in_two_percentage_global_sample
from Evaluation.Evaluator import EvaluatorHoldout



URM_all = sps.coo_matrix((URM_all_dataframe['data'], (URM_all_dataframe['user_id'], URM_all_dataframe['item_id'])))
URM_all = URM_all.tocsr()

URM_train_validation, URM_test = split_train_in_two_percentage_global_sample(URM_all, train_percentage = 0.8)
evaluator_test = EvaluatorHoldout(URM_test, cutoff_list=[10])

EvaluatorHoldout: Ignoring 141 ( 0.4%) Users that have less than 1 test interactions


In [4]:
from Recommenders.Neural.Convolution.GraphConvolution import GraphConvolutionWithBlockMatrix, GraphConvolution
import optuna

def objective_function(trial):
    factors = trial.suggest_int("factors", 10, 500)
    alpha = trial.suggest_float("alpha", 0, 15)
    recommender = GraphConvolution(URM_train_validation)
    recommender.fit(num_factors=factors, alpha=alpha, random_seed=42)
    
    result_dic, _ = evaluator_test.evaluateRecommender(recommender)
    MAP = result_dic.loc[10]["MAP"]
    return MAP

class SaveResults(object):
    
    def __init__(self):
        self.results_df = pd.DataFrame(columns = ["result"])
    
    def __call__(self, optuna_study, optuna_trial):
        hyperparam_dict = optuna_trial.params.copy()
        hyperparam_dict["result"] = optuna_trial.values[0]
        
        self.results_df = pd.concat([self.results_df, pd.DataFrame([hyperparam_dict])], ignore_index=True)
        self.results_df.to_csv("logs/GraphConvolution/gf_firstTrial.csv", index = False)

optuna_study = optuna.create_study(study_name="GraphConvolution", direction="maximize")
        
save_results = SaveResults()
        
optuna_study.optimize(objective_function,
                      callbacks=[save_results],
                      n_trials = 100,
                      )

[I 2024-12-30 17:21:59,170] A new study created in memory with name: GraphConvolution


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (498, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 3.15 min. Users per second: 188


[I 2024-12-30 17:32:45,073] Trial 0 finished with value: 0.034686656766933346 and parameters: {'factors': 498, 'alpha': 3.7139085869566326}. Best is trial 0 with value: 0.034686656766933346.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (91, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 3.82 min. Users per second: 155


[I 2024-12-30 17:44:33,645] Trial 1 finished with value: 0.020240066265771933 and parameters: {'factors': 91, 'alpha': 6.271988717134312}. Best is trial 0 with value: 0.034686656766933346.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (341, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 3.63 min. Users per second: 163


[I 2024-12-30 17:56:19,651] Trial 2 finished with value: 0.031164143795832537 and parameters: {'factors': 341, 'alpha': 10.78275386578148}. Best is trial 0 with value: 0.034686656766933346.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (60, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 3.07 min. Users per second: 193


[I 2024-12-30 18:07:07,451] Trial 3 finished with value: 0.016993923036274595 and parameters: {'factors': 60, 'alpha': 6.869817509996496}. Best is trial 0 with value: 0.034686656766933346.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (255, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 3.69 min. Users per second: 161


[I 2024-12-30 18:18:52,796] Trial 4 finished with value: 0.02862001864003479 and parameters: {'factors': 255, 'alpha': 14.37888527078544}. Best is trial 0 with value: 0.034686656766933346.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (100, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V


[W 2024-12-30 18:19:17,345] Trial 5 failed with parameters: {'factors': 100, 'alpha': 8.782889106263408} because of the following error: KeyboardInterrupt().
Traceback (most recent call last):
  File "/Users/lucapagano/miniconda3/envs/RecSysPD/lib/python3.8/site-packages/optuna/study/_optimize.py", line 197, in _run_trial
    value_or_values = func(trial)
  File "/var/folders/50/7hjr20g17_18p725nbt0qqv00000gn/T/ipykernel_35677/262824940.py", line 8, in objective_function
    recommender.fit(num_factors=factors, alpha=alpha, random_seed=42)
  File "/Users/lucapagano/Developer/RecSys/Competizione/Recommenders/Neural/Convolution/GraphConvolution.py", line 54, in fit
    VTDotV = self.V.T.dot(self.V)
  File "/Users/lucapagano/miniconda3/envs/RecSysPD/lib/python3.8/site-packages/scipy/sparse/_compressed.py", line 85, in __init__
    self._coo_container(arg1, dtype=dtype)
  File "/Users/lucapagano/miniconda3/envs/RecSysPD/lib/python3.8/site-packages/scipy/sparse/_coo.py", line 190, in __init

KeyboardInterrupt: 

# We can already see that we need bigger factors

In [ ]:
from Recommenders.Neural.Convolution.GraphConvolution import GraphConvolution
import optuna

def objective_function(trial):
    factors = trial.suggest_int("factors", 500, 2500)
    alpha = trial.suggest_float("alpha", 0, 15)
    recommender = GraphConvolution(URM_train_validation)
    recommender.fit(num_factors=factors, alpha=alpha, random_seed=42)
    
    result_dic, _ = evaluator_test.evaluateRecommender(recommender)
    MAP = result_dic.loc[10]["MAP"]
    return MAP

class SaveResults(object):
    
    def __init__(self):
        self.results_df = pd.DataFrame(columns = ["result"])
    
    def __call__(self, optuna_study, optuna_trial):
        hyperparam_dict = optuna_trial.params.copy()
        hyperparam_dict["result"] = optuna_trial.values[0]
        
        self.results_df = pd.concat([self.results_df, pd.DataFrame([hyperparam_dict])], ignore_index=True)
        self.results_df.to_csv("logs/GraphConvolution/gf_secondTrial.csv", index = False)

optuna_study = optuna.create_study(study_name="GraphConvolution", direction="maximize")
        
save_results = SaveResults()
        
optuna_study.optimize(objective_function,
                      callbacks=[save_results],
                      n_trials = 100,
                      )

[I 2024-12-30 18:40:58,341] A new study created in memory with name: GraphConvolution


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (1608, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 3.28 min. Users per second: 181


[I 2024-12-30 18:52:50,249] Trial 0 finished with value: 0.043327983241803945 and parameters: {'factors': 1608, 'alpha': 5.051769446290934}. Best is trial 0 with value: 0.043327983241803945.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2236, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 4.44 min. Users per second: 134


[I 2024-12-30 19:06:24,287] Trial 1 finished with value: 0.0454054553319184 and parameters: {'factors': 2236, 'alpha': 6.009407100047718}. Best is trial 1 with value: 0.0454054553319184.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2381, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 3.41 min. Users per second: 174


[I 2024-12-30 19:18:45,254] Trial 2 finished with value: 0.045802090092016766 and parameters: {'factors': 2381, 'alpha': 13.65337908117169}. Best is trial 2 with value: 0.045802090092016766.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (1960, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.75 min. Users per second: 216


[I 2024-12-30 19:30:12,873] Trial 3 finished with value: 0.044659397944689344 and parameters: {'factors': 1960, 'alpha': 13.232627727638812}. Best is trial 2 with value: 0.045802090092016766.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2302, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I


# Even higher

In [4]:
from Recommenders.Neural.Convolution.GraphConvolution import GraphConvolution
import optuna

def objective_function(trial):
    factors = trial.suggest_int("factors", 1000, 3500)
    alpha = trial.suggest_float("alpha", 0, 15)
    recommender = GraphConvolution(URM_train_validation)
    recommender.fit(num_factors=factors, alpha=alpha, random_seed=42)
    
    result_dic, _ = evaluator_test.evaluateRecommender(recommender)
    MAP = result_dic.loc[10]["MAP"]
    return MAP

class SaveResults(object):
    
    def __init__(self):
        self.results_df = pd.DataFrame(columns = ["result"])
    
    def __call__(self, optuna_study, optuna_trial):
        hyperparam_dict = optuna_trial.params.copy()
        hyperparam_dict["result"] = optuna_trial.values[0]
        
        self.results_df = pd.concat([self.results_df, pd.DataFrame([hyperparam_dict])], ignore_index=True)
        self.results_df.to_csv("logs/GraphConvolution/gf_thirdTrial.csv", index = False)

optuna_study = optuna.create_study(study_name="GraphConvolution", direction="maximize")
        
save_results = SaveResults()
        
optuna_study.optimize(objective_function,
                      callbacks=[save_results],
                      n_trials = 100,
                      )

[I 2024-12-30 19:40:20,841] A new study created in memory with name: GraphConvolution


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2341, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.83 min. Users per second: 210


[I 2024-12-30 19:52:35,151] Trial 0 finished with value: 0.04620460894944539 and parameters: {'factors': 2341, 'alpha': 0.7174368134138076}. Best is trial 0 with value: 0.04620460894944539.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (3033, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.75 min. Users per second: 215


[I 2024-12-30 20:05:09,533] Trial 1 finished with value: 0.04680404885651251 and parameters: {'factors': 3033, 'alpha': 7.23175879418644}. Best is trial 1 with value: 0.04680404885651251.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2002, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.68 min. Users per second: 221


[I 2024-12-30 20:17:08,336] Trial 2 finished with value: 0.0447459191477297 and parameters: {'factors': 2002, 'alpha': 11.037943396597363}. Best is trial 1 with value: 0.04680404885651251.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2187, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.70 min. Users per second: 220


[I 2024-12-30 20:28:47,159] Trial 3 finished with value: 0.04528229508781402 and parameters: {'factors': 2187, 'alpha': 11.258568316242709}. Best is trial 1 with value: 0.04680404885651251.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2623, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.66 min. Users per second: 223


[I 2024-12-30 20:40:07,299] Trial 4 finished with value: 0.04614409460932701 and parameters: {'factors': 2623, 'alpha': 12.037957655911061}. Best is trial 1 with value: 0.04680404885651251.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (1451, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.92 min. Users per second: 203


[I 2024-12-30 20:52:05,858] Trial 5 finished with value: 0.04276018122752072 and parameters: {'factors': 1451, 'alpha': 3.956042624788899}. Best is trial 1 with value: 0.04680404885651251.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (1427, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.82 min. Users per second: 211


[I 2024-12-30 21:03:10,297] Trial 6 finished with value: 0.04309368178605299 and parameters: {'factors': 1427, 'alpha': 1.2999532721318836}. Best is trial 1 with value: 0.04680404885651251.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (1739, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V


[W 2024-12-30 21:09:28,356] Trial 7 failed with parameters: {'factors': 1739, 'alpha': 2.669393926880328} because of the following error: KeyboardInterrupt().
Traceback (most recent call last):
  File "/Users/lucapagano/miniconda3/envs/RecSysPD/lib/python3.8/site-packages/optuna/study/_optimize.py", line 197, in _run_trial
    value_or_values = func(trial)
  File "/var/folders/50/7hjr20g17_18p725nbt0qqv00000gn/T/ipykernel_65980/2032058779.py", line 8, in objective_function
    recommender.fit(num_factors=factors, alpha=alpha, random_seed=42)
  File "/Users/lucapagano/Developer/RecSys/Competizione/Recommenders/Neural/Convolution/GraphConvolution.py", line 58, in fit
    partial_res = self.alpha * self.D_I_inv.dot(VTDotV)
  File "/Users/lucapagano/miniconda3/envs/RecSysPD/lib/python3.8/site-packages/scipy/sparse/_base.py", line 620, in __rmul__
    return self._rmul_dispatch(other)
  File "/Users/lucapagano/miniconda3/envs/RecSysPD/lib/python3.8/site-packages/scipy/sparse/_base.py", line

KeyboardInterrupt: 

# Ok we need more factors and run a longer trial all night

In [ ]:
from Recommenders.Neural.Convolution.GraphConvolution import GraphConvolution
import optuna

def objective_function(trial):
    factors = trial.suggest_int("factors", 2500, 7000)
    alpha = trial.suggest_float("alpha", 0, 15)
    recommender = GraphConvolution(URM_train_validation)
    recommender.fit(num_factors=factors, alpha=alpha, random_seed=42)
    
    result_dic, _ = evaluator_test.evaluateRecommender(recommender)
    MAP = result_dic.loc[10]["MAP"]
    return MAP

class SaveResults(object):
    
    def __init__(self):
        self.results_df = pd.DataFrame(columns = ["result"])
    
    def __call__(self, optuna_study, optuna_trial):
        hyperparam_dict = optuna_trial.params.copy()
        hyperparam_dict["result"] = optuna_trial.values[0]
        
        self.results_df = pd.concat([self.results_df, pd.DataFrame([hyperparam_dict])], ignore_index=True)
        self.results_df.to_csv("logs/GraphConvolution/gf_fourthTrial.csv", index = False)

optuna_study = optuna.create_study(study_name="GraphConvolution", direction="maximize")
        
save_results = SaveResults()
        
optuna_study.optimize(objective_function,
                      callbacks=[save_results],
                      n_trials = 100,
                      )

[I 2024-12-30 21:10:15,290] A new study created in memory with name: GraphConvolution


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2760, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.84 min. Users per second: 209


[I 2024-12-30 21:22:29,845] Trial 0 finished with value: 0.046335783739911714 and parameters: {'factors': 2760, 'alpha': 11.901829407351755}. Best is trial 0 with value: 0.046335783739911714.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2648, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.61 min. Users per second: 228


[I 2024-12-30 21:34:12,688] Trial 1 finished with value: 0.04633034000227268 and parameters: {'factors': 2648, 'alpha': 6.461344645519348}. Best is trial 0 with value: 0.046335783739911714.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5387, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.88 min. Users per second: 206


[I 2024-12-30 21:47:53,016] Trial 2 finished with value: 0.048184587633806286 and parameters: {'factors': 5387, 'alpha': 1.1654935837366893}. Best is trial 2 with value: 0.048184587633806286.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6198, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.85 min. Users per second: 208


[I 2024-12-30 22:02:48,671] Trial 3 finished with value: 0.04764093962724222 and parameters: {'factors': 6198, 'alpha': 4.114320645313396}. Best is trial 2 with value: 0.048184587633806286.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (4681, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.50 min. Users per second: 238


[I 2024-12-30 22:15:33,136] Trial 4 finished with value: 0.04776159818237178 and parameters: {'factors': 4681, 'alpha': 11.67845917608939}. Best is trial 2 with value: 0.048184587633806286.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6533, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.56 min. Users per second: 231


[I 2024-12-30 22:29:43,372] Trial 5 finished with value: 0.047798517046935086 and parameters: {'factors': 6533, 'alpha': 1.978297498436759}. Best is trial 2 with value: 0.048184587633806286.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2829, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.21 min. Users per second: 268


[I 2024-12-30 22:40:58,520] Trial 6 finished with value: 0.046571450868120194 and parameters: {'factors': 2829, 'alpha': 4.719744598202624}. Best is trial 2 with value: 0.048184587633806286.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (4305, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.53 min. Users per second: 234


[I 2024-12-30 22:53:27,216] Trial 7 finished with value: 0.047807484776931045 and parameters: {'factors': 4305, 'alpha': 11.256801910738595}. Best is trial 2 with value: 0.048184587633806286.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (4468, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.62 min. Users per second: 226


[I 2024-12-30 23:06:09,533] Trial 8 finished with value: 0.0478445675221891 and parameters: {'factors': 4468, 'alpha': 6.142636880635036}. Best is trial 2 with value: 0.048184587633806286.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (4441, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.66 min. Users per second: 223


[I 2024-12-30 23:18:44,680] Trial 9 finished with value: 0.0481383520959988 and parameters: {'factors': 4441, 'alpha': 1.320819566328479}. Best is trial 2 with value: 0.048184587633806286.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5713, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.41 min. Users per second: 246


[I 2024-12-30 23:32:06,033] Trial 10 finished with value: 0.048832784834679274 and parameters: {'factors': 5713, 'alpha': 0.1824647541257356}. Best is trial 10 with value: 0.048832784834679274.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5705, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.71 min. Users per second: 219


[I 2024-12-30 23:45:58,887] Trial 11 finished with value: 0.048809268512384985 and parameters: {'factors': 5705, 'alpha': 0.33531406722615265}. Best is trial 10 with value: 0.048832784834679274.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5645, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.56 min. Users per second: 232


[I 2024-12-30 23:59:26,825] Trial 12 finished with value: 0.04890634385514104 and parameters: {'factors': 5645, 'alpha': 0.2233286445252573}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5467, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.61 min. Users per second: 228


[I 2024-12-31 00:12:49,043] Trial 13 finished with value: 0.047731356062580624 and parameters: {'factors': 5467, 'alpha': 14.650086861160336}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6921, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.58 min. Users per second: 230


[I 2024-12-31 00:27:45,316] Trial 14 finished with value: 0.047294103416520565 and parameters: {'factors': 6921, 'alpha': 3.8801803112094064}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5992, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.62 min. Users per second: 226


[I 2024-12-31 00:41:45,626] Trial 15 finished with value: 0.04761470199354536 and parameters: {'factors': 5992, 'alpha': 8.186821895838984}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (3667, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.71 min. Users per second: 219


[I 2024-12-31 00:53:37,921] Trial 16 finished with value: 0.047534936688537935 and parameters: {'factors': 3667, 'alpha': 2.7630011150163485}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5102, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.62 min. Users per second: 226


[I 2024-12-31 01:06:49,647] Trial 17 finished with value: 0.04780441340744593 and parameters: {'factors': 5102, 'alpha': 8.287712875612108}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (3826, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.45 min. Users per second: 242


[I 2024-12-31 01:18:39,984] Trial 18 finished with value: 0.04825817006579574 and parameters: {'factors': 3826, 'alpha': 0.2869091167734327}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6391, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.55 min. Users per second: 233


[I 2024-12-31 01:32:50,318] Trial 19 finished with value: 0.0476621426676191 and parameters: {'factors': 6391, 'alpha': 2.938561262870884}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (4968, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.55 min. Users per second: 233


[I 2024-12-31 01:45:45,707] Trial 20 finished with value: 0.04784647611912501 and parameters: {'factors': 4968, 'alpha': 9.581492357338604}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5892, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.58 min. Users per second: 230


[I 2024-12-31 01:59:21,547] Trial 21 finished with value: 0.04888697583261246 and parameters: {'factors': 5892, 'alpha': 0.2523354389138704}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5645, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.55 min. Users per second: 233


[I 2024-12-31 02:12:43,121] Trial 22 finished with value: 0.04858843648898241 and parameters: {'factors': 5645, 'alpha': 0.12654194678627112}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5942, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.66 min. Users per second: 223


[I 2024-12-31 02:26:19,974] Trial 23 finished with value: 0.04790086555762729 and parameters: {'factors': 5942, 'alpha': 2.2366536704173776}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6907, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.57 min. Users per second: 230


[I 2024-12-31 02:40:50,848] Trial 24 finished with value: 0.04718400791978357 and parameters: {'factors': 6907, 'alpha': 5.326753021479133}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5203, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.66 min. Users per second: 223


[I 2024-12-31 02:54:10,214] Trial 25 finished with value: 0.048074192246545966 and parameters: {'factors': 5203, 'alpha': 3.21094839209093}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5885, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.60 min. Users per second: 228


[I 2024-12-31 03:07:38,410] Trial 26 finished with value: 0.048102199122846134 and parameters: {'factors': 5885, 'alpha': 1.3317747007923801}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6519, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.75 min. Users per second: 216


[I 2024-12-31 03:22:44,085] Trial 27 finished with value: 0.04884622305165736 and parameters: {'factors': 6519, 'alpha': 0.14035363845612828}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6643, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.55 min. Users per second: 233


[I 2024-12-31 03:37:20,829] Trial 28 finished with value: 0.047789681982263514 and parameters: {'factors': 6643, 'alpha': 1.5851362089519698}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6252, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.73 min. Users per second: 217


[I 2024-12-31 03:51:36,870] Trial 29 finished with value: 0.04737079066303508 and parameters: {'factors': 6252, 'alpha': 13.951770343799778}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6490, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.95 min. Users per second: 201


[I 2024-12-31 04:06:44,045] Trial 30 finished with value: 0.04764162636539187 and parameters: {'factors': 6490, 'alpha': 3.5092197812024986}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6103, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.74 min. Users per second: 216


[I 2024-12-31 04:20:35,398] Trial 31 finished with value: 0.04889757902505286 and parameters: {'factors': 6103, 'alpha': 0.20374150595551627}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6141, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.61 min. Users per second: 227


[I 2024-12-31 04:34:19,685] Trial 32 finished with value: 0.04836538705944325 and parameters: {'factors': 6141, 'alpha': 0.8573910796269615}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6682, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.64 min. Users per second: 225


[I 2024-12-31 04:48:37,334] Trial 33 finished with value: 0.04752820531686763 and parameters: {'factors': 6682, 'alpha': 2.232066187354219}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5503, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.69 min. Users per second: 220


[I 2024-12-31 05:01:56,973] Trial 34 finished with value: 0.0484348468328646 and parameters: {'factors': 5503, 'alpha': 0.8405678019017273}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6111, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.58 min. Users per second: 230


[I 2024-12-31 05:15:51,967] Trial 35 finished with value: 0.04792683897550995 and parameters: {'factors': 6111, 'alpha': 2.191764662956837}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5251, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.50 min. Users per second: 237


[I 2024-12-31 05:28:43,799] Trial 36 finished with value: 0.046831180587605166 and parameters: {'factors': 5251, 'alpha': 0.0348500332724946}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6358, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.63 min. Users per second: 225


[I 2024-12-31 05:42:49,795] Trial 37 finished with value: 0.047622075509979205 and parameters: {'factors': 6358, 'alpha': 4.526309301356008}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6722, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.68 min. Users per second: 222


[I 2024-12-31 05:57:11,949] Trial 38 finished with value: 0.047835383514269345 and parameters: {'factors': 6722, 'alpha': 1.408329353094081}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (4897, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.24 min. Users per second: 265


[I 2024-12-31 06:09:41,418] Trial 39 finished with value: 0.04794008767059583 and parameters: {'factors': 4897, 'alpha': 5.379952604653593}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5863, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.51 min. Users per second: 237


[I 2024-12-31 06:22:53,544] Trial 40 finished with value: 0.04762227729505259 and parameters: {'factors': 5863, 'alpha': 6.506713715642308}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5677, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.28 min. Users per second: 260


[I 2024-12-31 06:36:03,068] Trial 41 finished with value: 0.04851746165526002 and parameters: {'factors': 5677, 'alpha': 0.8159553719061026}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5408, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.57 min. Users per second: 231


[I 2024-12-31 06:49:04,647] Trial 42 finished with value: 0.04844929397520889 and parameters: {'factors': 5408, 'alpha': 0.7806941257830429}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5744, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.30 min. Users per second: 258


[I 2024-12-31 07:02:00,655] Trial 43 finished with value: 0.04803832801557003 and parameters: {'factors': 5744, 'alpha': 1.6685821645881511}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6176, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.69 min. Users per second: 220


[I 2024-12-31 07:15:57,960] Trial 44 finished with value: 0.04779517254295843 and parameters: {'factors': 6176, 'alpha': 2.7593937997724733}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (4753, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.67 min. Users per second: 222


[I 2024-12-31 07:28:48,963] Trial 45 finished with value: 0.0482838335596429 and parameters: {'factors': 4753, 'alpha': 0.11275403117361506}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (2870, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.42 min. Users per second: 245


[I 2024-12-31 07:40:09,638] Trial 46 finished with value: 0.0467005933150039 and parameters: {'factors': 2870, 'alpha': 2.147757565232023}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5553, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.61 min. Users per second: 227


[I 2024-12-31 07:53:23,646] Trial 47 finished with value: 0.048427304976397204 and parameters: {'factors': 5553, 'alpha': 0.8921893615392886}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6040, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.64 min. Users per second: 224


[I 2024-12-31 08:07:12,654] Trial 48 finished with value: 0.04848889736163025 and parameters: {'factors': 6040, 'alpha': 0.6712387598174137}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6310, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.43 min. Users per second: 244


[I 2024-12-31 08:20:46,439] Trial 49 finished with value: 0.047503450413267416 and parameters: {'factors': 6310, 'alpha': 9.839561928229386}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6832, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.23 min. Users per second: 265


[I 2024-12-31 08:35:02,225] Trial 50 finished with value: 0.04732642135844643 and parameters: {'factors': 6832, 'alpha': 3.618881536199168}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5827, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.54 min. Users per second: 234


[I 2024-12-31 08:48:22,233] Trial 51 finished with value: 0.048231182148373644 and parameters: {'factors': 5827, 'alpha': 0.08198609547687374}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5387, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.42 min. Users per second: 245


[I 2024-12-31 09:01:15,281] Trial 52 finished with value: 0.04808497269769732 and parameters: {'factors': 5387, 'alpha': 1.5606827034738497}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6505, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.07 min. Users per second: 287


[I 2024-12-31 09:15:09,597] Trial 53 finished with value: 0.04856897816484647 and parameters: {'factors': 6505, 'alpha': 0.48869173438793123}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5678, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.65 min. Users per second: 224


[I 2024-12-31 09:28:34,083] Trial 54 finished with value: 0.047944824603061244 and parameters: {'factors': 5678, 'alpha': 2.625856889580654}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (6001, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
EvaluatorHoldout: Processed 35595 (100.0%) in 2.12 min. Users per second: 279


[I 2024-12-31 09:41:34,362] Trial 55 finished with value: 0.048061401748505 and parameters: {'factors': 6001, 'alpha': 1.2393675283284893}. Best is trial 12 with value: 0.04890634385514104.


GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5225, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V


# We can stop here, we can start the submission
I must test if having low factors despite a worse validation is less overfitted and more beneficial to the hybrid!  

Top 10 is 


| Rank | Result       | Factors | Alpha     |
|------|--------------|---------|-----------|
| 1    | 0.048906     | 5645.0  | 0.223329  |
| 2    | 0.048898     | 6103.0  | 0.203742  |
| 3    | 0.048887     | 5892.0  | 0.252335  |
| 4    | 0.048846     | 6519.0  | 0.140354  |
| 5    | 0.048833     | 5713.0  | 0.182465  |
| 6    | 0.048809     | 5705.0  | 0.335314  |
| 7    | 0.048588     | 5645.0  | 0.126542  |
| 8    | 0.048569     | 6505.0  | 0.488692  |
| 9    | 0.048517     | 5677.0  | 0.815955  |
| 10   | 0.048489     | 6040.0  | 0.671239  |
| 11   | 0.048449     | 5408.0  | 0.780694  |
| 12   | 0.048435     | 5503.0  | 0.840568  |
| 13   | 0.048427     | 5553.0  | 0.892189  |
| 14   | 0.048365     | 6141.0  | 0.857391  |
| 15   | 0.048284     | 4753.0  | 0.112754  |
| 16   | 0.048258     | 3826.0  | 0.286909  |
| 17   | 0.048231     | 5827.0  | 0.081986  |
| 18   | 0.048185     | 5387.0  | 1.165494  |
| 19   | 0.048138     | 4441.0  | 1.320820  |
| 20   | 0.048102     | 5885.0  | 1.331775  |



We can see that entry 16 despite having a sensible lower amount of num factors had a similar MAP. 

Let's first do a submission with the best parameters and then one with entry 16

In [13]:
from Recommenders.Neural.Convolution.GraphConvolution import GraphConvolution

best_params = {'num_factors' : 5645, 'alpha' : 0.223329, 'random_seed':42}

recommender = GraphConvolution(URM_all)
recommender.fit(**best_params)

target_user = pd.read_csv("Dataset/data_target_users_test.csv")
print(target_user.head(10))

user_id = target_user['user_id']
recommendations = []
for user in user_id:
    recommendations.append(recommender.recommend(user, cutoff=10))

# Saving in a submission file for the competition
submission = pd.DataFrame(columns=['user_id', 'item_list'])
submission['user_id'] = user_id
# Convert list of recommendations into a space-separated string
submission['item_list'] = [' '.join(map(str, recs)) for recs in recommendations]

# Save the submission file
num_factors = best_params["num_factors"]
alpha = best_params["alpha"]
submission.to_csv(f"submissions/graphConvolution_num_factors:{num_factors}_alpha:_{alpha}.csv", index=False)

GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5645, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
   user_id
0        0
1        1
2        2
3        3
4        4
5        5
6        6
7        7
8        8
9        9


In [14]:
W_sparse = recommender.W_sparse

sps.save_npz(f'graphConvolution_W_sparse_num_factors:{num_factors}_alpha:_{alpha}.npz', W_sparse)

# Let's also save the W_sparse with only train and validation for hybrid

In [15]:
import gc
gc.collect()

0

In [16]:
from Recommenders.Neural.Convolution.GraphConvolution import GraphConvolution

best_params = {'num_factors' : 5645, 'alpha' : 0.223329, 'random_seed':42}

recommender = GraphConvolution(URM_train_validation)
recommender.fit(**best_params)

W_sparse = recommender.W_sparse

sps.save_npz(f'graphConvolution_W_sparse_TRAINVAL_num_factors:{num_factors}_alpha:_{alpha}.npz', W_sparse)

GraphConvolution: Computing SVD decomposition of the normalized adjacency matrix...
Computing first term
D_I_inv shape: (38121, 38121)
V shape: (5645, 38121)
D_I shape: (38121, 38121)
Computing second term: V^T * V
Computing third term: alpha * D_I_inv * V^T * V
Computing fourth term: third_term * D_I
Computed Similarity matrix
